### PR TITLE
Added largest distance between nodes in a tree in C++

### LINKED ISSUE #4453: Largest distance between nodes of a tree
## 💥 Proposal

Hi I am a GSSOC participant,
Given an unweighted rooted tree which consists of N nodes, the goal of the problem is to find the largest distance between two nodes in a tree.
I would like to contribue the code for this problem in C++.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### PR DESCRIPTION
**Pull request by sg-iitg**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added the code for "Largest distance between two nodes in a tree" in C++ language following all guidelines

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #4453

### CODE REVIEW

**Comment by sg-iitg** [comment]
@HarshCasper @siddharth25pandey @ankitaggarwal23 
Please review this Pull Request
Thankyou!

File: C-Plus-Plus/graphs/Largest_distance_between_2_nodes_in_a_tree.cpp [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+The approach is to use DFS here, first we will find the node which is at farthest distance from the root node by DFS
+After this we will take this new node and run DFS from it to the farthest node.
+The distance between these 2 nodes is the answer
+It can be proven that the node found to be farthest from root node, has to be one of the end points of the largest distance path*/
+
+#include <bits/stdc++.h>
+using namespace std;
+
+//We maintain a visited array to keep track of what all nodes have already been visited
+//And an adjacency list(adj) can be formed using the given input array(A)
```
Comment (HarshCasper) [inline C-Plus-Plus/graphs/Largest_distance_between_2_nodes_in_a_tree.cpp]: Use Multi-Line Comments

File: C-Plus-Plus/graphs/Largest_distance_between_2_nodes_in_a_tree.cpp [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+int visited[40010];
+vector<int> adj[40010];
+vector<int> A;
+
+//mx stores the answer at all times
+int mx;
+
+//DFS function runs a DFS from the source node (node), and keeps updating the max distance until all nodes are visited
+//nd stores the node which is found to be at farthest distance from the source node (node)
```
Comment (HarshCasper) [inline C-Plus-Plus/graphs/Largest_distance_between_2_nodes_in_a_tree.cpp]: Use Multi-Line Comments

File: C-Plus-Plus/README.md [hunk 3]
```diff
@@ -196,6 +196,7 @@
 - [In-order (LNR) Tree Traversal](graphs/inorder-traversal.cpp)
 - [Kosaraju Algorithm](graphs/Kosaraju_Algorithm.cpp)
 - [Kruskal's Algorithm](graphs/kruskal_Algorithm.cpp)
+- [Largest distance between 2 nodes of a tree](graphs/Largest_distance_between_2_nodes_in_a_tree.cpp)
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: Keep it `Tree` instead of `tree`

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by HarshCasper — APPROVED** [review]

**Comment by HarshCasper** [comment]
Please make an issue here and raise a PR, for an article as part of our Docs repository: https://github.com/TesseractCoding/NeoAlgo-Docs

It would be much appreciated. Thanks!

**Pull request merged**